core/src/main/java/com/yahoo/oak/NoFreeMemoryManager.java

```diff
@@ -70,7 +71,7 @@ public boolean decodeReference(Slice s, long reference) {
             // key's slice is updated upon reference decoding.
             // Slice is not invalidated between next iterator steps and all the rest information
             // in slice remains the same.
-            s.setOffsetAndLength(rcd.getSecond(reference), rcd.getThird(reference));
+            s.setIterationOnSameBlock(rcd.getSecond(reference), rcd.getThird(reference));
```

Comment:
`setIterationOnSameBlock()` is unclear. Maybe `updateOnSameBlock()`?

Reply:
fixed